src/write.rs

```diff
@@ -476,7 +483,177 @@ where
                 Absorb::absorb_second(w_inner, op, &*r_handle);
             }
         } else {
-            self.oplog.extend(ops);
+            // Only try to compress if it is enabled, else use efficient fallback.
+            if T::MAX_COMPRESS_RANGE > 0 {
+                // Compress oplog by rev-iterating all ops appended since the last publish
+                // while attempting to combine them with the next op,
+                // cut short when an attempt fails due to encountering a dependency (e.g. clear then set).
+
+                // used to avoid walking more of the oplog than necessary.
+                let mut rev_dirty_range = 0usize..0;
+                for mut next in ops {
+                    // used to avoid linear insertion time in case of predominantly independent ops.
+                    let mut range_remaining = T::MAX_COMPRESS_RANGE;
+                    // used to more efficiently insert next if possible
+                    let mut none: Option<(usize, &mut Option<O>)> = None;
+                    // rev-iterate all unpublished and potentially non-none ops already in the oplog
+                    for (prev_rev_idx, prev_loc) in {
```

Comment:
Instead of doing the sanity checking inline in the iterator, let's just have a separate `for` loop behind `#[cfg(test)]` above that checks all the relevant indicates, and then have _this_ `for` loop just do the non-test version.

Reply:
Also note that `#[cfg(test)]` won't apply for integration tests, only unit tests. It may be that we want `#[cfg(debug_assertions)]` here instead.

Reply:
Done: 03823329b34bf1f3503c9a0e082033ef86d22f3c